lib/cdo/active_job_backend.rb

```diff
@@ -0,0 +1,281 @@
+require 'cdo/chat_client'
+require 'delayed/command'
+
+module Cdo
+  module ActiveJobBackend
+    # Restarts delayed_job workers in a rolling fashion, to prevent downtime.
+    def self.restart_workers(n_workers_to_start:, rolling_restart_in_n_batches: 1)
+      pid = fork do
+        # pre-cache Rails environment so each `delayed_job` worker command invocation
+        # doesn't take N minutes on production.
+        before_worker_fork
+        restart_workers_internal(n_workers_to_start, n_batches: rolling_restart_in_n_batches)
+      end
+      Process.wait(pid)
+    end
+
+    def self.restart_workers_internal(n_workers_to_start, n_batches:, start_time: Time.now, sigkill_timeout_s: 60.seconds)
```

Comment:
I like the increase from 30 to 60 seconds because a substantial number of EvaluateRubricJob executions fall into this range.

Reply:
Want to go higher? We're gonna end timing out and SIGKILLing at least some workers on almost every batch because there'll probably always be at least one aichat worker in each batch.

Thus, more or less: `worker deploy time = n_batch * sigkill_timeout_s`
Currently on prod: 5 minutes (5 batches * 60)

Reply:
what makes you think that aichat workers will need to be sigkilled? I would expect that after receiving the TERM signal, a worker will finish its current job and then terminate before picking up a new one, but that's just my opinion of how the worker should be implemented I suppose

Reply:
60s seems high enough to me because very few EvaluateRubricJobs take longer than this.

Reply:
> what makes you think that aichat workers will need to be sigkilled?

My brain was not working in a way I cannot even begin to explain: somehow I compared 1 minute with 5s and concluded that 5s was 5x bigger because 5 is 5x bigger than 1.